### ISSUE TITLE
Permission to use Node and Python Releases from node-versions and python-versions repositories?

### ISSUE THREAD

**Issue by MarioRivis** [actions/virtual-environments#3630]
Hello,

I would like to build a CLI tool that installs Node and Python runtime environments on personal computers/laptops. To do this I would like to take advantage of the already existing node and python distributions that are publicly available in the [node-versions](https://github.com/actions/node-versions) and [python-versions](https://github.com/actions/python-versions) repositories.

Although the license of the repository is MIT, I found the following caution message on each repo:
> Caution: this is prepared for and only permitted for use by actions virtual-environments and setup-node action.

Since my tool has absolutely no relation to the Github Actions environment and since all the resources are publically available, is it ok if I rely on the release assets containing the node and python distributions and download them with the CLI tool?

Thanks in advance!

**Comment by catthehacker** [comment]
I would recommend not relying on actions team repos but official ones that provide more versions and more architectures.
For Node you can grab latest tags from https://github.com/nodejs/node/releases or https://nodejs.org/download/release/ (which also contains pre-packaged releases for most architectures and OSes).
Python releases are hosted here: https://www.python.org/ftp/python/

**Comment by MarioRivis** [comment]
Thank you @catthehacker! Will do so.

**Comment by catthehacker** [comment]
Not sure in what language you are doing that but I recently had to deal with that in pure `bash`
Here is small snippet that grabs latest version specific by `NODE_VERSION` environmental variable, you will need to account for different architecture and OS in `"https://nodejs.org/download/release/latest-v${NODE_VERSION}.x/node-${ver}-linux-x64.tar.xz"`
```bash
ver=$(curl https://nodejs.org/download/release/index.json | jq "[.[] | select(.version|test(\"^v${NODE_VERSION}\"))][0].version" -r) \
&& node_path=$AGENT_TOOLSDIRECTORY/node/$(echo $ver | sed 's/v//g')/x64 \
&& mkdir -v -m 0777 -p $node_path \
&& curl "https://nodejs.org/download/release/latest-v${NODE_VERSION}.x/node-${ver}-linux-x64.tar.xz" | tar -Jxf - --strip-components=1 -C $node_path \
```

**Comment by maxim-lobanov** [comment]
@MarioRivis , 
For Node.js, `actions/node-versions` just contains official packages from https://nodejs.org/download/release/ so you can use them directly, as @catthehacker mentioned.
For Python on Windows, the same like Node.js.
For Python on Ubuntu, there is no official binaries as far as I know, so we build Python from source code and distribute it via `actions/python-versions`. You can try testing them on your local machine but we can't really guarantee that these binaries will work as expected outside of GitHub Actions because it wasn't our goal initially and have never tested them this way.

**Comment by MarioRivis** [comment]
Thank you, @maxim-lobanov!
I also don't know of any official python releases for Ubuntu. I found this repository: https://github.com/indygreg/python-build-standalone interesting, but I think they also build the distributions from code. I only tested it for windows and it worked ok.

Thanks for the answer, I will try to use the releases from python-versions for ubuntu & mac and fall back to the above mentioned repo if needed.
